Makefile

```diff
@@ -180,9 +180,8 @@ build.init: $(UP)
 # Setup Terraform for fetching provider schema
 TERRAFORM := $(TOOLS_HOST_DIR)/terraform-$(TERRAFORM_VERSION)
 TERRAFORM_WORKDIR := $(WORK_DIR)/terraform
-TERRAFORM_CLI_CONFIG_FILE := $(TERRAFORM_WORKDIR)/.terraformrc-custom
-TERRAFORM_FILESYSTEM_MIRROR := $(TOOLS_HOST_DIR)/$(TERRAFORM_PROVIDER_RELEASE)
-TERRAFORM_PROVIDER := $(TERRAFORM_FILESYSTEM_MIRROR)/registry.terraform.io/$(TERRAFORM_PROVIDER_SOURCE)/$(TERRAFORM_PROVIDER_VERSION)/$(SAFEHOST_PLATFORM)/terraform-provider-aws
+# The terraform aws provider hardcodes version 99.99.99 for local builds
+TERRAFORM_PROVIDER := $(ROOT_DIR)/upstream/terraform-plugin-dir/registry.terraform.io/$(TERRAFORM_PROVIDER_SOURCE)/99.99.99/$(SAFEHOST_PLATFORM)/terraform-provider-aws
```

Comment:
I think still preserving the $(TERRAFORM_PROVIDER_VERSION) here makes more sense to me. Especially in terms of keeping multiple versions of the build in the local environment and making it look consistent.

But here I have a question let's consider there are more than one binary for different versions after my suggested change, then will the `-plugin-dir` flag confuse to use which binary?

Reply:
The challenge with that approach is patching the terraform provider's own build process to write the binary to a path that contains the desired version. I suppose we could also do it by copying the artifact to a path that we choose in the work dir, not the upstream submodule, but I found the simplicity of this approach appealing. 

Conceptually, we aren't using any published version of the tf provider any more, but rather one built from patched source code. So the hard coded 99.99.99 seemed fine to me. 

When working on this and the provider version upgrade, I was fairly frequently switching back and forth between the two tf provider versions, and rebuilding the tf provider. It was faster than I expected.

Reply:
Also important to note: this path is precisely where `make go-build` in the terraform provider puts the compiled binary.